substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ProgressReporter.java

```diff
@@ -740,8 +739,24 @@ private static Timer getTimer(TimerCollection.Registry type) {
         return TimerCollection.singleton().get(type);
     }
 
+    private static void installShutdownHook(Runnable hook) {
+        try {
+            Runtime.getRuntime().addShutdownHook(new Thread(hook));
+        } catch (IllegalStateException e) {
+            /* If the VM is already shutting down, we do not need to register shutdownHook. */
+        }
+    }
+
     private static void resetANSIMode() {
-        NativeImageSystemIOWrappers.singleton().getOut().print(ANSI.RESET);
+        var out = NativeImageSystemIOWrappers.singleton().getOut();
+        out.print(ANSI.RESET);
+        out.flush();
+    }
+
+    private static void ringTerminalBell() {
+        var out = NativeImageSystemIOWrappers.singleton().getOut();
+        out.write(0x07);
```

Comment:
Does this work on Windows?

Reply:
Yes, at least the command prompt supports the bell character.